test/Writer/Factory/WriterFactoryTest.php

```diff
@@ -0,0 +1,290 @@
+        $container = $this->getMockForAbstractClass(ServiceLocatorInterface::class);
+        $factory = new WriterFactory();
+
+        // Act
+        $object = $factory($container, InvokableObject::class, []);
+
+        // Assert
+        $this->assertInstanceOf(InvokableObject::class, $object);
+        $this->assertEquals([], $object->options);
+    }
+
+    /**
+     * @covers \Zend\Log\Writer\Factory\WriterFactory::__invoke
+     */
+    public function testInvokeWithInvalidFilterManagerAsString()
+    {
+        // Arrange
+        $container = $this->getMockForAbstractClass(ContainerInterface::class);
+        $factory = new WriterFactory();
+
+        // Act
+        $object = $factory($container, InvokableObject::class, [
+            'filter_manager' => 'my_manager',
+        ]);
+
+        // Assert
+        $this->assertInstanceOf(InvokableObject::class, $object);
+        $this->assertEquals([
+            'filter_manager' => null,
+        ], $object->options);
+    }
+
+    /**
+     * @covers \Zend\Log\Writer\Factory\WriterFactory::__invoke
+     */
+    public function testInvokeWithValidFilterManagerAsString()
+    {
+        // Arrange
+        $container = $this->getMockForAbstractClass(ContainerInterface::class);
+        $container->expects($this->once())->method('get')->with($this->equalTo('my_manager'))->willReturn(123);
+
+        $factory = new WriterFactory();
+
+        // Act
+        $object = $factory($container, InvokableObject::class, [
+            'filter_manager' => 'my_manager',
+        ]);
+
+        // Assert
+        $this->assertInstanceOf(InvokableObject::class, $object);
+        $this->assertEquals([
+            'filter_manager' => 123,
+        ], $object->options);
+    }
+
+    /**
+     * @covers \Zend\Log\Writer\Factory\WriterFactory::__invoke
+     */
+    public function testInvokeWithoutFilterManager()
+    {
+        // Arrange
+        $container = $this->getMockForAbstractClass(ContainerInterface::class);
+        $container->expects($this->at(0))->method('has')->with($this->equalTo('LogFilterManager'))->willReturn(true);
+        $container->expects($this->at(1))->method('get')->with($this->equalTo('LogFilterManager'))->willReturn(123);
+
+        $factory = new WriterFactory();
+
+        // Act
+        $object = $factory($container, InvokableObject::class, []);
+
+        // Assert
+        $this->assertInstanceOf(InvokableObject::class, $object);
+        $this->assertEquals([
+            'filter_manager' => 123,
+        ], $object->options);
+    }
+
+    /**
+     * @covers \Zend\Log\Writer\Factory\WriterFactory::__invoke
+     */
+    public function testInvokeWithInvalidFormatterManagerAsString()
+    {
+        // Arrange
+        $container = $this->getMockForAbstractClass(ContainerInterface::class);
+        $factory = new WriterFactory();
+
+        // Act
+        $object = $factory($container, InvokableObject::class, [
+            'formatter_manager' => 'my_manager',
```

Comment:
Same comment as for the filter manager; invalid string should raise an exception, as it's an invalid service.